extensions/mongodb/src/main/java/com/hazelcast/jet/mongodb/impl/ReadMongoP.java

```diff
@@ -250,10 +250,8 @@ void connect(MongoClient newClient, boolean snapshotsEnabled) {
 
         abstract void restore(Object value);
 
-        void close() {
-            if (mongoClient != null) {
-                mongoClient.close();
-            }
+        @Override
+        public void close() {
```

Comment:
I would move this `close()` implementaiton to `BatchMongoReader` to make it obvious that implementation has to take care of closing its resources

Reply:
Done